### PR TITLE
Backport #15788 to v3.0.1-rhel

### PR DESCRIPTION
**Pull request by mheon**
Cherry-pick #15788 to v3.0.1-rhel branch per RHBZ 2157930

As part of this, did a separate `git grep` pass to ensure that no instances of GenerateNonCryptoID existed in 3.0.1 that were not present when the original patch was made, which caught around 5 extra occurrances, mostly in tests.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mheon** [comment]
@containers/podman-maintainers PTAL

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
Addresses https://bugzilla.redhat.com/show_bug.cgi?id=2157930

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM, assuming happy tests.

**Comment by mheon** [comment]
One test failing; there's a missing tag on quay.io for an alpine image.

I'm pretty sure this is unrelated, and caused by how old the tests on this branch are.

**Comment by mheon** [comment]
Can't find the actual fix upstream, so I'm going to disable the test.

**Comment by rhatdan** [comment]
I will smash it through.

**Pull request merged**
